server/Server.js

```diff
@@ -253,6 +253,10 @@ class Server {
     // if RouterBasePath is set, modify all requests to include the base path
     if (global.RouterBasePath) {
       app.use((req, res, next) => {
+        const host = req.get('host')
+        const protocol = req.secure || req.get('x-forwarded-proto') === 'https' ? 'https' : 'http'
+        const prefix = req.url.startsWith(global.RouterBasePath) ? global.RouterBasePath : ''
+        req.originalHostPrefix = `${protocol}://${host}${prefix}`
```

Comment:
We need to define this even if `global.RouterBasePath` is not set.
Testing this without setting the `RouterBasePath` results in undefined passed to `buildXml`
![image](https://github.com/user-attachments/assets/7d90405f-8ca5-4d12-bffb-b2ce3b2ebcef)

Reply:
Sorry about this. I made some changes after switching to non-empty ROUTER_BASE_PATH and haven't properly checked them. This is fixed now.

_Note_: `urlStartsWithRouterBasePath` is true if `global.RouterBasePath` is empty, so the I removed the `if (global.RouterBasePath)` check altogether